Userland/Applications/PixelPaint/Tools/MoveTool.cpp

```diff
@@ -121,6 +121,11 @@ void MoveTool::on_keydown(GUI::KeyEvent& event)
     if (!layer)
         return;
 
+    if (m_layer_being_moved.is_null() && event.key() == KeyCode::Key_Escape) {
+        m_editor->image().selection().clear();
+        return;
+    }
```

Comment:
Perhaps we should handle the escape key at a higher level instead of making tools care about it.
Wouldn't it make sense for escape to clear the selection no matter which tool is selected?

Reply:
Definitely makes sense, I'll look into that.  One thing to note though is that I think tools still need to be able to receive Escape, for example the Text Tool uses Escape to cancel out of adding some text.

Maybe have Image Editor check for Escape key before it passes the event to the tool, and in the case where there is a selection, ImageEditor clears it.

Reply:
Latest push moves this logic to ImageEditor and modifies tools that actually make use of the Escape in certain contexts to mark the KeyEvent as "ignore". Upon return from the tool's keydown handler, Image Editor will check if the event is accepted still and if so process its Escape logic.

This allows for Escape to me used during Polygon selection and Lasso selection to cancel an in-progress selection, without clearing any already existing selected region.

Reply:
@awesomekling per chat in #ui channel on Discord it seems like I'm not using `accept()`/`ignore()` how they're intended, and this is in part due to a bug in PixelPaint's Tool class, where it doesn't properly call `ignore()` on the event if nothing used it. My understanding is that the fix to this Tool class is in the PR about inverting the shortcut/key event processing. Which includes various fixups like this to properly `ignore()` events.

I'm happy to wait for that to land and rebase/refactor this before we move it farther along if you agree.

Reply:
or I can implement that Tool fix and adjust the required PixelPaint tools enough to get them using it properly enough for this PR to continue and the fix could be dropped from the other PR.

Reply:
I've rebased this on latest master which includes the fixes I needed (PR #15788). I've adjusted adjusted the implementation here to handle the "escape clears selection" at the ImageEditor level.